cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java

```diff
@@ -0,0 +1,476 @@
+
+        private final Map<String, String> map;
+        private BufferedReader bufferedReader;
+        private int peekChar = NO_CHAR;
+
+        public PropertiesReader(Map<String, String> map) {
+            this.map = map;
+        }
+
+        /**
+         * Read map values from a String.
+         *
+         * @param keyValuePairs String containing name=value pairs separated by newlines.
+         */
+        public void read(String keyValuePairs) {
+            try {
+                read(new StringReader(keyValuePairs));
+            } catch (IOException e) {
+                throw new IllegalStateException("StringReader should not cause IOException", e);
+            }
+        }
+
+        /**
+         * Read map values from a Reader.  The Reader is closed after all values are read.
+         *
+         * @param reader Reader containing name=value pairs separated by newlines.
+         * @throws IOException when read or close fails
+         */
+        public void read(Reader reader) throws IOException {
+            try (BufferedReader bufferedReader = new BufferedReader(reader)) {
+                read(bufferedReader);
+            }
+        }
+
+        /**
+         * Read map values from a BufferedReader.  The BufferedReader is not closed after all values are
+         * read.
+         *
+         * @param bufferedReader Reader containing name=value pairs separated by newlines.
+         * @throws IOException when read or close fails
+         */
+        public void read(BufferedReader bufferedReader) throws IOException {
+            this.bufferedReader = bufferedReader;
+            for (; ; ) {
+                ignoreWhitespace(true);
+                if (peekChar == EOF) {
+                    return;
+                }
+
+                if (isComment()) {
+                    ignoreToEndOfLine();
+                    continue;
+                }
+
+                String key = readToken(true);
+
+                ignoreWhitespace(false);
+                if (isKeySeparator()) {
+                    peekChar = NO_CHAR;
+                    ignoreWhitespace(false);
+                }
+
+                String value = readToken(false);
+                map.put(key, value);
+            }
+        }
+
+        private String readToken(boolean breakOnKeySeperator) throws IOException {
+            StringBuilder sb = new StringBuilder();
+            for (; ; ) {
+                switch (peekChar) {
+                    case '\t':
+                    case '\f':
+                    case ' ':
+                    case ':':
+                    case '=':
+                        if (!breakOnKeySeperator) {
+                            break;
+                        }
+                    case '\r':
+                    case '\n':
+                    case EOF:
+                        return sb.toString();
+                    case '\\':
+                        sb.append(readEscape());
```

Comment:
`Properties.load()` includes the documentation:

> If a logical line is spread across several natural lines, the
> backslash escaping the line terminator sequence, the line
> terminator sequence, and any white space at the start of the
> following line have no affect on the key or element values.

AFAICS this doesn't deal with the "any white space at the start of the following line" part. Could you at least add this to the tests?

Reply:
I have added a couple of unit tests for this case.  I did need to make a minor change.